app/src/main/java/com/example/android/trackr/MainActivity.kt

```diff
@@ -50,12 +50,12 @@ class MainActivity : AppCompatActivity() {
         // cannot do that properly since the parent receives the insets first and will consume some
         // of them. We'll remove the rail's built-in listener and apply the padding ourselves.
         binding.activityRoot.doOnApplyWindowInsets { v, insets, _, _ ->
-            if (binding.navigationRail.isGone) {
+            if (binding.navigationRail?.isGone == true) {
```

Comment:
Since we only need `doOnApplyWindowInsets` if there's a navigation rail, maybe we can lift this up. Also, since we're no longer using visibility for the rail, we can just do a null check.
```
binding.navigationRail?.let { navRail ->
    binding.activityRoot.doOnApplyWindowInsets { v, insets, _, _, ->
        val isRtl = v.isRtl
        ...
    }
}
```